app/assets/javascripts/provisioning_template.js

```diff
@@ -136,8 +154,35 @@ function set_diff_mode(item){
   session.setValue(patch);
 }
 
+function set_render_mode() {
+  Editor.setTheme("ace/theme/twilight");
+  Editor.setReadOnly(true);
+  var session = Editor.getSession();
+  session.setMode("ace/mode/text");
+  $(session).off('change');
+  get_rendered_template();
+}
+
+function get_rendered_template(){
+  var session = Editor.getSession();
+  host_id = $("#preview_host_selector select").val();
+  url = $('.template_text').data('render-path');
+  template = $('#new').val();
+
+  session.setValue(__('Rendering the template, please wait...'));
+  $.post(url, { template: template, preview_host_id: host_id }, function(response) {
+    $("div#preview_error").hide();
+    $("div#preview_error span.text").html('');
+    session.setValue(response);
+  }).fail(function(response){
+    $("div#preview_error span.text").html(response.responseText);
+    $("div#preview_error").show();
+    session.setValue(__('There was an error during rendering, fix it in code and retry'));
```

Comment:
"There was an error during rendering, return to the Code tab to edit the template."?  ("Fix it in code" sounds odd.)